modules/gateways/monero/library.php

```diff
@@ -330,18 +344,102 @@ public function transfer($amount_xmr, $address, $mixin = 4)
+	 * @param  string   $payment_id       Payment ID                                           (optional)
+	 * @param  number   $mixin            Mixin number (ringsize - 1)                          (optional)
+	 * @param  number   $priority         Payment ID                                           (optional)
+	 * @param  number   $below_amount     Only send outputs below this amount                  (optional)
+	 * @param  number   $unlock_time      UNIX time or block height to unlock output           (optional)
+	 * @param  boolean  $do_not_relay     Do not relay transaction                             (optional)
+	 *
+	 *   OR
+	 *
+	 * @param  object  $params            Array containing any of the options listed above, where only address is required
+	 *
+	 * @return object  Example: {
+	 *   "amount": "1000000000000",
+	 *   "fee": "1000020000",
+	 *   "tx_hash": "c60a64ddae46154a75af65544f73a7064911289a7760be8fb5390cb57c06f2db",
+	 *   "tx_key": "805abdb3882d9440b6c80490c2d6b95a79dbc6d1b05e514131a91768e8040b04"
+	 * }
+	 *
+	 */
+	public function sweep_all($address, $subaddr_indices = '', $account_index = 0, $payment_id = '', $mixin = 10, $priority = 2, $below_amount = 0, $unlock_time = 0, $do_not_relay = false)
+	{
+		if (is_array($address)) { // Parameters passed in as object/dictionary
+			$params = $address;
+
+			if (array_key_exists('address', $params)) {
+				$address = $params['address'];
+			} else {
+				throw new Exception('Error: Address required');
+			}
+			if (array_key_exists('subaddr_indices', $params)) {
+				$subaddr_indices = $params['subaddr_indices'];
+			}
+			if (array_key_exists('account_index', $params)) {
+				$account_index = $params['account_index'];
+			}
+			if (array_key_exists('payment_id', $params)) {
+				$payment_id = $params['payment_id'];
+			}
+			if (array_key_exists('mixin', $params)) {
+				$mixin = $params['mixin'];
+			}
+			if (array_key_exists('priority', $params)) {
+				$priority = $params['priority'];
+			}
+			if (array_key_exists('below_amount', $params)) {
+				$below_amount = $params['below_amount'];
+			}
+			if (array_key_exists('unlock_time', $params)) {
+				$unlock_time = $params['unlock_time'];
+			}
+			if (array_key_exists('do_not_relay', $params)) {
+				$do_not_relay = $params['do_not_relay'];
+			}
+		}
+
+		$params = array('address' => $address, 'mixin' => $mixin, 'get_tx_key' => true, 'subaddr_indices' => $subaddr_indices, 'account_index' => $account_index, 'payment_id' => $payment_id, 'priority' => $priority, 'below_amount' => $this->_transform($below_amount), 'unlock_time' => $unlock_time, 'do_not_relay' => $do_not_relay);
+		$sweep_all_method = $this->_run('sweep_all', $params);
+
+		$save = $this->store(); // Save wallet state after transfer
+
+		return $sweep_all_method;
+	}
+
+	/**
+	 *
+	 * Save wallet
+	 *
+	 * @param  none
+	 *
+	 * @return object  Example:
+	 *
+	 */
+	public function store()
+	{
+		return $this->_run('store');
+	}
+
```

Comment:
Snatched this code from the other library which I noticed has more functions. 
https://github.com/monero-integrations/monerophp/blob/master/src/walletRPC.php

We may want to consider pulling that whole file over here and updating any calls to use it so its easier to reuse that upstream file